webapp/components/_new/generic/UserProfileHeader/UserProfileHeader.vue

```diff
@@ -0,0 +1,244 @@
+      return 'Profile header image of ' + this.user.name
+    },
+    profileHeaderUrl() {
+      if (this.user == null || this.user.profileHeader == null) return false
+      const { profileHeader } = this.user
+      return profileHeader && profileHeader.url
+    },
+    headerStyles() {
+      /*  Allow the header to shrink when a profileHeaderUrl is present,
+          this prevents a grey bar from showing up on mobile.
+      */
+      if (this.profileHeaderUrl) {
+        return {
+          // Still setting some height in case the user uploads a super super wide picture.
+          // In that case the height would, in theory, be able to go to 0 thus changing a profile
+          // picture won't be possible anymore.
+          '--min-height': 20 + 'px',
+          '--max-height': 250 + 'px',
+        }
+      } else {
+        // If no picture is present, the height is fixed to 250 so user can click the upload zone.
+        return {
+          '--height': 250 + 'px',
+        }
+      }
+    },
+  },
+  watch: {
+    error() {
+      const that = this
+      setTimeout(function () {
+        that.error = false
+      }, 2000)
+    },
+  },
+  methods: {
+    template() {
+      return `<div class="dz-preview dz-file-preview">
+                <div class="dz-image">
+                  <div data-dz-thumbnail-bg></div>
+                </div>
+              </div>
+      `
+    },
+    vddrop(file) {
+      const profileHeaderUpload = file[0]
+      this.$apollo
+        .mutate({
+          mutation: updateUserMutation(),
+          variables: {
+            profileHeader: {
+              upload: profileHeaderUpload,
+            },
+            id: this.user.id,
+          },
+        })
+        .then(() => {
+          this.$toast.success(this.$t('user.profileHeader.submitted'))
+        })
+        .catch((error) => this.$toast.error(error.message))
+    },
+    verror(file, message) {
+      if (file.status === 'error') {
+        this.error = true
+        this.$toast.error(file.status, message)
+      }
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+.profile-header {
+  min-height: var(--min-height);
+  max-height: var(--max-height);
+  height: var(--height);
+  background-color: DarkGrey; /* Fallback color */
+}
+
+.profile-header-image {
+  width: 100%;
+}
+
+#profileHeaderDropzone {
+  height: 100%;
+}
+
+.dz-message {
+  height: 100%;
+}
+
+#profileHeaderDropzone .dz-preview {
+  transition: all 0.2s ease-out;
+  width: 160px;
+  display: flex;
+}
+
+#profileHeaderDropzone .dz-preview .dz-image {
+  width: 100%;
+  height: 100%;
+  object-fit: contain;
+  overflow: hidden;
+}
+
+#profileHeaderDropzone .dz-preview .dz-image > div {
+  width: inherit;
+  height: inherit;
+  border-radius: 50%;
+  background-size: cover;
+}
+
+#profileHeaderDropzone .dz-preview .dz-image > img {
+  width: 100%;
+}
+
+.profileHeader-attachments-upload-area {
+  height: 100%;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-06-25T11:52:46Z" title="Thursday, June 25th 2020, 1:52:46 pm +02:00">Jun 25, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -169,7 +164,10 @@ export default {
 
 .profileHeader-attachments-upload-area {
   height: 100%;
-  position: relative;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
```


The last changes here and/or in `webapp/pages/profile/_id/_slug.vue` made the whole header image area and dropZone invisible, unfortunately.

I haven’t investigated why. 😬

Reply:
> [<img alt="DriesCruyskens" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/DriesCruyskens) **Authored by [DriesCruyskens](https://github.com/DriesCruyskens)**
_<time datetime="2020-06-25T16:27:38Z" title="Thursday, June 25th 2020, 6:27:38 pm +02:00">Jun 25, 2020</time>_

---


I yes, I went a little too fast there. Was so focused on getting the profile header right I forgot to check how it looked when no header present.. my bad. Should all be good now (if tests succeed). proof:

![fix](https://user-images.githubusercontent.com/26900172/85759630-6aa8dd00-b711-11ea-9766-3ff1c5d2a013.gif)

I'm using CSS variables with different values depending on the presence of the header image. 

- Header image: allow a small shrink to prevent the grey bar using min- and max-height. 
- No header image: fixed height of 250px;

When uploading tho you can see a short visible glitch/shrink of the header. I tried setting a fallback height of 250px in css (for when the css variables don't come through) but this didn't help. Any idea what causes this or how to fix it?